diff -ru a/src/utils.c b/src/utils.c
--- a/src/utils.c	2023-09-28 00:30:12.800167716 +0100
+++ b/src/utils.c	2023-09-28 00:35:37.230694025 +0100
@@ -3348,12 +3348,9 @@
   }
 
   #ifndef MINGW
-  	char *directorio_home;
-  	directorio_home=getenv("HOME");
-  	if ( directorio_home==NULL) {
-                  //printf("Unable to find $HOME environment variable to open configuration file\n");
-  		return 0;
-  	}
+  	char *envpath = getenv("HOME");
+  	const char *cpath = ("/ArchyPie/configs/zesarux/");
+  	const char *directorio_home = strcat(envpath, cpath);
 
   	sprintf(configfile,"%s/%s",directorio_home,DEFAULT_ZESARUX_CONFIG_FILE);
 
diff -ru a/src/utils.h b/src/utils.h
--- a/src/utils.h	2023-09-28 00:30:12.801167737 +0100
+++ b/src/utils.h	2023-09-28 00:36:27.354684259 +0100
@@ -636,7 +636,7 @@
 
 extern int util_tape_tap_get_info(z80_byte *tape,char *texto,int origin_tap);
 
-#define DEFAULT_ZESARUX_CONFIG_FILE ".zesaruxrc"
+#define DEFAULT_ZESARUX_CONFIG_FILE "zesaruxrc"
 
 extern char *customconfigfile;
 
