diff -ru a/src/utils.c b/src/utils.c
--- a/src/utils.c	2024-06-15 20:36:23.117574051 +0100
+++ b/src/utils.c	2024-06-20 15:14:40.005194618 +0100
@@ -3371,14 +3371,17 @@
   }
 
   #ifndef MINGW
-  	char *directorio_home;
-  	directorio_home=getenv("HOME");
-  	if ( directorio_home==NULL) {
-                  //printf("Unable to find $HOME environment variable to open configuration file\n");
-  		return 0;
-  	}
+        char *chome;
+        char *cpath;
+        chome = getenv("HOME");
+        cpath = ("ArchyPie/configs/zesarux");
+
+        if (chome==NULL) {
+                //printf("Unable to find $HOME environment variable to open configuration file\n");
+                return 0;
+        }
 
-  	sprintf(configfile,"%s/%s",directorio_home,DEFAULT_ZESARUX_CONFIG_FILE);
+        sprintf(configfile,"%s/%s/%s",chome,cpath,DEFAULT_ZESARUX_CONFIG_FILE);
 
   #else
   	char *homedrive;
@@ -22648,4 +22651,4 @@
     }
 
     return 1;
-}
\ No newline at end of file
+}
diff -ru a/src/utils.h b/src/utils.h
--- a/src/utils.h	2024-06-15 20:36:23.117574051 +0100
+++ b/src/utils.h	2024-06-15 20:47:03.687149454 +0100
@@ -644,7 +644,7 @@
 
 extern int util_tape_tap_get_info(z80_byte *tape,char *texto,int origin_tap);
 
-#define DEFAULT_ZESARUX_CONFIG_FILE ".zesaruxrc"
+#define DEFAULT_ZESARUX_CONFIG_FILE "zesaruxrc"
 
 extern char *customconfigfile;
 
