diff -ru a/res/config.xml b/res/config.xml
--- a/res/config.xml	2022-07-27 17:22:16.915589000 +0100
+++ b/res/config.xml	2022-08-02 13:14:08.772270816 +0100
@@ -4,12 +4,15 @@
     External Data Required
 -->
 <data>
-    <!-- OutRun Revision B ROM directory -->
+    <!-- OutRun Revision B ROM directory 
     <rompath>roms/</rompath>
-    <!-- Path to load required binary files from -->
+    -->
+    <!-- Path to load required binary files from 
     <respath>./res/</respath>
-    <!-- Path to write XML for scores -->
+    -->
+    <!-- Path to write XML for scores 
     <savepath>./</savepath>
+    -->
     <!-- Use CRC 32 Based ROM Loading (i.e. rely on scanning file contents, not on filenames) -->
     <crc32>1</crc32>
 </data>
@@ -24,7 +27,7 @@
          1 is full-screen
          2 is full-screen stretched, no borders. (Aspect ratio will be lost. Scanlines will be disabled.)
     -->
-    <mode>0</mode>
+    <mode>1</mode>
     <!-- 
         Windowed Mode Settings 
         Scale: Scale or zoom factor for window mode. 1 is original size. 
diff -ru a/src/main/frontend/config.cpp b/src/main/frontend/config.cpp
--- a/src/main/frontend/config.cpp	2022-07-27 17:22:16.955590000 +0100
+++ b/src/main/frontend/config.cpp	2022-08-02 16:18:39.995106795 +0100
@@ -36,8 +36,16 @@
 
 Config::Config(void)
 {
-    data.cfg_file = "./config.xml";
-    
+    char* envcpath = (char *) malloc(100);
+    char* cpath = (char *) malloc(100);
+
+    strcpy(envcpath, getenv("HOME"));
+    strcpy(cpath, "/.config/archypie/ports/cannonball/config.xml");
+
+    const char* arpie_cpath = strcat(envcpath, cpath);
+
+    data.cfg_file = arpie_cpath;
+
     // Setup default sounds
     music_t magical, breeze, splash;
     magical.title = "MAGICAL SOUND SHOWER";
@@ -87,9 +95,23 @@
     // ------------------------------------------------------------------------
     // Data Settings
     // ------------------------------------------------------------------------
-    data.rom_path         = pt_config.get("data.rompath", "roms/");  // Path to ROMs
-    data.res_path         = pt_config.get("data.respath", "res/");   // Path to ROMs
-    data.save_path        = pt_config.get("data.savepath", "./");    // Path to Save Data
+
+    char* envrpath = (char *) malloc(100);
+    char* envspath = (char *) malloc(100);
+    char* rpath = (char *) malloc(100);
+    char* spath = (char *) malloc(100);
+
+    strcpy(envrpath, getenv("HOME"));
+    strcpy(envspath, getenv("HOME"));
+    strcpy(rpath, "/ArchyPie/roms/ports/cannonball/");
+    strcpy(spath, "/ArchyPie/roms/ports/cannonball/hiscores");
+
+    const char* arpie_rpath = strcat(envrpath, rpath);
+    const char* arpie_spath = strcat(envspath, spath);
+
+    data.rom_path         = pt_config.get("data.rompath", arpie_rpath);         // Path to ROMs
+    data.res_path         = pt_config.get("data.respath", "./res/");            // Path to Resources
+    data.save_path        = pt_config.get("data.savepath", arpie_spath);        // Path to Save Data
     data.crc32            = pt_config.get("data.crc32", 1);
 
     data.file_scores      = data.save_path + "hiscores.xml";
@@ -550,4 +572,4 @@
     }
     else
         engine.dip_traffic++;
-}
\ No newline at end of file
+}
