--- a/gtk/src/gtk_config.cpp	2024-07-10 22:28:19.304093574 +0100
+++ b/gtk/src/gtk_config.cpp	2024-07-10 23:33:25.591682333 +0100
@@ -26,25 +26,10 @@
 {
     // Find config directory
     char *env_home = getenv("HOME");
-    char *env_xdg_config_home = getenv("XDG_CONFIG_HOME");
-
-    if (!env_home && !env_xdg_config_home)
-    {
-        return std::string{".snes9x"};
-    }
 
     fs::path config;
-    // If XDG_CONFIG_HOME is set, use that, otherwise guess default
-    if (env_xdg_config_home)
-    {
-        config = env_xdg_config_home;
-        config /= "snes9x";
-    }
-    else
-    {
-        config = env_home;
-        config /= ".config/snes9x";
-    }
+    config = env_home;
+    config /= "ArchyPie/configs/snes9x";
 
     if (!fs::exists(config))
         fs::create_directories(config);
