--- a/Makefile	Sun Sep  8 03:07:37 2024
+++ b/Makefile	Mon Nov 11 20:16:45 2024
@@ -52,9 +52,6 @@
 #FE_HWACCEL_VAAPI=1
 #FE_HWACCEL_VDPAU=1
 #
-# If you set this to 1, AM+ will link the system SFML version
-# If left to blank, AM+ will build and link the extlib/SFML version
-#USE_SYSTEM_SFML=1
 ###############################
 
 #FE_DEBUG=1
@@ -78,7 +75,6 @@
 RM=rm -f
 MD=mkdir -p
 WINDRES=windres
-CMAKE=cmake
 LD=objcopy
 B64FLAGS = -w0
 
@@ -89,8 +85,6 @@
 OPTIMIZE=2
 endif
 
-STATIC ?= 1
-
 ifndef VERBOSE
  SILENT=@
  CC_MSG = @echo Compiling $@...
@@ -103,7 +97,6 @@
 override CC := $(TOOLCHAIN)-$(CC)
 override CXX := $(TOOLCHAIN)-$(CXX)
 override AR := $(TOOLCHAIN)-$(AR)
-override CMAKE := $(TOOLCHAIN)-$(CMAKE)
 override LD := $(TOOLCHAIN)-$(LD)
 endif
 
@@ -258,13 +251,23 @@
 #
 # Deal with SFML
 #
-ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
-SFML_OBJ_DIR = $(OBJ_DIR)/sfml
-SFML_LIB_DIR=$(SFML_OBJ_DIR)/install/lib/
-SFML_PKG_CONFIG_PATH=$(ROOT_DIR)/$(SFML_OBJ_DIR)/install/pkgconfig
-LIBS += -L$(SFML_LIB_DIR)
 SFML_PC="sfml-system sfml-window sfml-graphics"
-SFML_TOKEN=$(SFML_OBJ_DIR)/.sfmlok
+ifeq ($(STATIC),1)
+  LIBS += $(shell $(PKG_CONFIG) --static --libs-only-L $(SFML_PC))
+  $(info Manually adding sfml libs as pkg-config has no --static version)
+  LIBS += -lsfml-graphics-s -lsfml-window-s -lsfml-system-s
+  CFLAGS += -DSFML_STATIC $(shell $(PKG_CONFIG) --static --cflags $(SFML_PC))
+  ifeq ($(FE_WINDOWS_COMPILE),1)
+  else ifeq ($(FE_MACOSX_COMPILE),1)
+  else
+    LIBS += -lGL -lGLU -lm -lz -ludev -lrt
+  endif
+else
+  # SFML may not generate .pc files, so manually add libs
+  CFLAGS += $(shell $(PKG_CONFIG) --cflags $(SFML_PC))
+  LIBS += $(shell $(PKG_CONFIG) --libs $(SFML_PC))
+  #LIBS += -lsfml-graphics -lsfml-window -lsfml-system
+endif
 
 ifeq ($(FE_MACOSX_COMPILE),1)
   LIBS += -framework OpenGL
@@ -431,6 +434,9 @@
  SQUIRREL += $(OBJ_DIR)/libnowide.a
 endif
 
+$(info flags:$(CFLAGS) $(FE_FLAGS))
+$(info libs:$(LIBS))
+
 OBJ = $(patsubst %,$(OBJ_DIR)/%,$(_OBJ))
 DEP = $(patsubst %,$(SRC_DIR)/%,$(_DEP))
 RES = $(patsubst %,$(OBJ_DIR)/%.h,$(_RES))
@@ -484,73 +490,12 @@
 
 .PHONY: clean
 .PHONY: install
-.PHONY: sfml sfmlbuild
-.PHONY: bin2h
-
-SFML_FLAGS =
-ifneq ($(USE_SYSTEM_SFML),1)
-sfmlbuild:
-ifneq ("$(wildcard $(SFML_TOKEN))","")
-	$(info SFML is already built)
-else
-	$(info Building SFML...)
-ifeq ($(STATIC),1)
-	$(eval SFML_FLAGS += -DBUILD_SHARED_LIBS=FALSE)
-ifeq ($(FE_WINDOWS_COMPILE),1)
-	$(eval SFML_FLAGS += -DCMAKE_CXX_FLAGS="-DAL_LIBTYPE_STATIC=TRUE")
-endif
-else
-	$(eval SFML_FLAGS += -DBUILD_SHARED_LIBS=TRUE)
-endif
-ifeq ($(USE_DRM),1)
-	$(eval SFML_FLAGS += -DSFML_USE_DRM=1)
-endif
-ifeq ($(FE_MACOSX_COMPILE),1)
-	$(eval SFML_FLAGS += -DSFML_USE_SYSTEM_DEPS=1)
-endif
-	$(SILENT)$(CMAKE) -S extlibs/SFML -B $(SFML_OBJ_DIR) -DCMAKE_INSTALL_PREFIX=$(SFML_OBJ_DIR)/install -DOpenGL_GL_PREFERENCE=GLVND -DSFML_INSTALL_PKGCONFIG_FILES=TRUE -DSFML_BUILD_NETWORK=FALSE $(SFML_FLAGS)
-	+$(SILENT)$(CMAKE) --build obj/sfml --config Release --target install
-	touch $(SFML_TOKEN)
-endif
-else
-sfmlbuild:
-
-endif
-
-sfml: sfmlbuild
-ifeq ($(STATIC),1)
-	$(eval SFML_LIBS += $(shell PKG_CONFIG_PATH$(PKG_CONFIG_MXE)="$(SFML_PKG_CONFIG_PATH)" $(PKG_CONFIG) --static --libs-only-L $(SFML_PC)))
-	$(info Manually adding sfml libs as pkg-config has no --static version)
-	$(eval SFML_LIBS += -lsfml-graphics-s -lsfml-window-s -lsfml-system-s)
-	$(eval CFLAGS += -DSFML_STATIC $(shell PKG_CONFIG_PATH$(PKG_CONFIG_MXE)="$(SFML_PKG_CONFIG_PATH)" $(PKG_CONFIG) --static --cflags $(SFML_PC)))
-ifeq ($(FE_WINDOWS_COMPILE),1)
-else ifeq ($(FE_MACOSX_COMPILE),1)
-else
-	$(eval SFML_LIBS += -lGL -lGLU -lm -lz -ludev -lrt)
-endif
-else
-	# SFML may not generate .pc files, so manually add libs
-	$(eval CFLAGS += $(shell PKG_CONFIG_PATH$(PKG_CONFIG_MXE)="$(SFML_PKG_CONFIG_PATH)" $(PKG_CONFIG) --cflags $(SFML_PC)))
-	$(eval SFML_LIBS += $(shell PKG_CONFIG_PATH$(PKG_CONFIG_MXE)="$(SFML_PKG_CONFIG_PATH)" $(PKG_CONFIG) --libs $(SFML_PC)))
-	#LIBS += -lsfml-graphics -lsfml-window -lsfml-system
-endif
-	$(eval override LIBS = $(SFML_LIBS) $(LIBS))
-
-# .WAIT is supported from make 4.4, not yet standard sadly. So the all target appreared
-#$(OBJ_DIR) : sfml .WAIT headerinfo
-$(OBJ_DIR): headerinfo
-	$(MD) $@
 
 $(RES_FONTS_DIR): $(OBJ_DIR)
 	$(MD) $@
 
 $(RES_IMGS_DIR): $(OBJ_DIR)
 	$(MD) $@
-
-headerinfo: sfml
-	$(info flags:$(CFLAGS) $(FE_FLAGS))
-	$(info libs:$(LIBS))
-
 #
 # Expat Library
 #
@@ -626,7 +571,7 @@
 	$(CC_MSG)
 	$(SILENT)$(CXX) -c $< -o $@ $(CFLAGS) $(SQUIRREL_FLAGS)
 
-$(SQSTDLIB_OBJ_DIR): sfml
+$(SQSTDLIB_OBJ_DIR):
 	$(MD) $@
 
 #
@@ -649,7 +594,7 @@
 	$(CC_MSG)
 	$(SILENT)$(CXX) -c $< -o $@ $(CFLAGS)
 
-$(AUDIO_OBJ_DIR): sfml
+$(AUDIO_OBJ_DIR):
 	$(MD) $@
 
 #
