--- a/trunk/src/CGame.cpp	2022-05-29 17:10:52.465856000 +0100
+++ b/trunk/src/CGame.cpp	2022-08-02 16:25:57.848396620 +0100
@@ -322,7 +322,9 @@
 #ifdef WIN32
         const char* appDataPath = getenv("APPDATA");
 #else
-        const char* appDataPath = getenv("HOME");
+        char* env_var = getenv("HOME");
+        const char* path = "/.config/archypie/ports/bombermaaan";
+        const char* appDataPath = strcat(env_var, path);
 #endif
         if (!appDataPath)
         {
@@ -353,7 +355,7 @@
             }
         }
 #else
-        dynamicDataFolder.append("/.Bombermaaan/");
+        dynamicDataFolder;
         struct stat dirStatus;
 
         if (stat(dynamicDataFolder.c_str(), &dirStatus) == 0 && S_ISDIR(dirStatus.st_mode))
