--- a/trunk/src/COptions.cpp	2022-05-29 13:37:58.000000000 
+++ b/trunk/src/COptions.cpp	2022-05-29 17:46:17.000000000 
@@ -207,18 +207,14 @@
     // Number of matches for a battle
     m_BattleCount = 3;
 
     // First level file (index=0) is selected
     m_Level = 0;
 
-    // Default display mode is windowed, not full-screen
-#ifdef _DEBUG
-    m_DisplayMode = DISPLAYMODE_WINDOWED;
-#else
+    // Default display mode is full-screen
     m_DisplayMode = DISPLAYMODE_FULL3;
-#endif
 
     // Set the bomber types
     m_BomberType[0] = BOMBERTYPE_MAN;
     m_BomberType[1] = BOMBERTYPE_COM;
     m_BomberType[2] = BOMBERTYPE_OFF;
     m_BomberType[3] = BOMBERTYPE_OFF;
