diff -ru a/source/common/platform/posix/unix/i_specialpaths.cpp b/source/common/platform/posix/unix/i_specialpaths.cpp
--- a/source/common/platform/posix/unix/i_specialpaths.cpp	2023-04-04 18:22:03.424070815 +0100
+++ b/source/common/platform/posix/unix/i_specialpaths.cpp	2023-04-04 18:32:46.757631423 +0100
@@ -110,9 +110,7 @@
 
 FString M_GetAppDataPath(bool create)
 {
-	// Don't use GAME_DIR and such so that ZDoom and its child ports can
-	// share the node cache.
-	FString path = NicePath("$HOME/.config/" GAMENAMELOWERCASE);
+	FString path = NicePath("$HOME/" GAME_DIR);
 	if (create)
 	{
 		CreatePath(path);
@@ -130,9 +128,7 @@
 
 FString M_GetCachePath(bool create)
 {
-	// Don't use GAME_DIR and such so that ZDoom and its child ports can
-	// share the node cache.
-	FString path = NicePath("$HOME/.config/zdoom/cache");
+	FString path = NicePath("$HOME/" GAME_DIR "/cache");
 	if (create)
 	{
 		CreatePath(path);

diff -ru a/source/core/version.h b/source/core/version.h
--- a/source/core/version.h	2023-04-04 18:22:03.529072622 +0100
+++ b/source/core/version.h	2023-04-04 18:45:07.657700297 +0100
@@ -86,7 +86,7 @@
 #if defined(__APPLE__) || defined(_WIN32)
 #define GAME_DIR GAMENAMELOWERCASE
 #else
-#define GAME_DIR ".config/" GAMENAMELOWERCASE
+#define GAME_DIR "ArchyPie/configs/" GAMENAMELOWERCASE
 #endif
 
 #define DEFAULT_DISCORD_APP_ID "954282576464449556"
