diff -u a/def.h b/def.h
--- a/def.h	2022-08-05 11:56:08.177997000 +0100
+++ b/def.h	2022-08-05 12:39:02.275086938 +0100
@@ -82,7 +82,7 @@
 #define PATH_MAX 1024
 #endif
 #endif
-#define ININAME strncat(strncpy(malloc(PATH_MAX),getenv("HOME"),PATH_MAX),"/.digger.rc",PATH_MAX)
+#define ININAME strncat(strncpy(malloc(PATH_MAX),getenv("HOME"),PATH_MAX),"/.config/archypie/ports/digger/digger.rc",PATH_MAX)
 #else
 #define ININAME "DIGGER.INI"
 #endif
diff -u a/scores.c b/scores.c
--- a/scores.c	2022-08-05 11:56:08.181997000 +0100
+++ b/scores.c	2022-08-05 12:39:52.331014803 +0100
@@ -58,7 +58,7 @@
 
 #elif defined UNIX && !defined _VGL
 
-#define SFNAME strncat(strncpy(malloc(PATH_MAX),getenv("HOME"),PATH_MAX),"/.digger.sco",PATH_MAX)
+#define SFNAME strncat(strncpy(malloc(PATH_MAX),getenv("HOME"),PATH_MAX),"/.config/archypie/ports/digger/digger.sco",PATH_MAX)
 
 #else
 
