diff -ru a/rott/rt_datadir.c b/rott/rt_datadir.c
--- a/rott/rt_datadir.c	2024-02-25 01:25:28.993830559 +0000
+++ b/rott/rt_datadir.c	2024-02-25 03:00:02.139079387 +0000
@@ -53,21 +53,10 @@
         char *result;
 
 #ifndef _WIN32
-        result = SDL_GetPrefPath("", PACKAGE_TARNAME);
-        if (result != NULL)
-        {
-            dir = M_StringDuplicate(result);
-            SDL_free(result);
-        }
-        else
+        char *homedir = getenv("HOME");
+        dir = M_StringJoin(homedir, "/ArchyPie/configs/taradino", PATH_SEP_STR, NULL);
 #endif
-        {
-            result = GetExeDir();
-            dir = M_StringDuplicate(result);
-        }
-
         M_MakeDirectory(dir);
-
 #if !(SHAREWARE == 1)
         result = dir;
         dir = M_StringJoin(result, "darkwar", PATH_SEP_STR, NULL);
@@ -140,20 +129,12 @@
             homedir = "/";
         }
 
-        tmp_env = M_StringJoin(homedir, "/.local/share", NULL);
+        tmp_env = M_StringJoin(homedir, "/ArchyPie/roms/ports", NULL);
         env = tmp_env;
     }
 
-    AddDataDir(M_StringJoin(env, "/games/rott", NULL));
+    AddDataDir(M_StringJoin(env, "/rott", NULL));
     free(tmp_env);
-
-    env = getenv("XDG_DATA_DIRS");
-    if (env == NULL)
-    {
-        env = "/usr/local/share:/usr/share";
-    }
-
-    AddDataPath(env, "/games/rott");
 }
 #endif
 
