--- a/mupen64plus-core/src/osal/files_unix.c	2024-07-11 11:12:03.875669460 +0100
+++ b/mupen64plus-core/src/osal/files_unix.c	2024-07-11 10:42:40.763954973 +0100
@@ -174,13 +174,8 @@
     static char retpath[PATH_MAX];
     int rval;
 
-    /* first, try the XDG_CONFIG_HOME environment variable */
-    rval = get_xdg_dir(retpath, "XDG_CONFIG_HOME", "mupen64plus/");
-    if (rval == 0)
-        return retpath;
-
-    /* then try the HOME environment variable */
-    rval = get_xdg_dir(retpath, "HOME", ".config/mupen64plus/");
+    /* Use the HOME environment variable */
+    rval = get_xdg_dir(retpath, "HOME", "ArchyPie/configs/mupen64plus/");
     if (rval == 0)
         return retpath;
 
@@ -195,13 +190,8 @@
     static char retpath[PATH_MAX];
     int rval;
 
-    /* first, try the XDG_DATA_HOME environment variable */
-    rval = get_xdg_dir(retpath, "XDG_DATA_HOME", "mupen64plus/");
-    if (rval == 0)
-        return retpath;
-
-    /* then try the HOME environment variable */
-    rval = get_xdg_dir(retpath, "HOME", ".local/share/mupen64plus/");
+    /* Use the HOME environment variable */
+    rval = get_xdg_dir(retpath, "HOME", "ArchyPie/configs/mupen64plus/");
     if (rval == 0)
         return retpath;
 
