diff -ru a/pcsx2/Pcsx2Config.cpp b/pcsx2/Pcsx2Config.cpp
--- a/pcsx2/Pcsx2Config.cpp	2023-10-30 00:49:45.671892556 +0000
+++ b/pcsx2/Pcsx2Config.cpp	2023-10-30 01:05:53.479260500 +0000
@@ -1728,26 +1728,10 @@
 		CoTaskMemFree(documents_directory);
 	}
 #elif defined(__linux__) || defined(__FreeBSD__)
-	// Use $XDG_CONFIG_HOME/PCSX2 if it exists.
-	const char* xdg_config_home = getenv("XDG_CONFIG_HOME");
+	const char* xdg_config_home = getenv("HOME");
 	if (xdg_config_home && Path::IsAbsolute(xdg_config_home))
 	{
-		DataRoot = Path::Combine(xdg_config_home, "PCSX2");
-	}
-	else
-	{
-		// Use ~/PCSX2 for non-XDG, and ~/.config/PCSX2 for XDG.
-		// Maybe we should drop the former when Qt goes live.
-		const char* home_dir = getenv("HOME");
-		if (home_dir)
-		{
-			// ~/.config should exist, but just in case it doesn't and this is a fresh profile..
-			const std::string config_dir(Path::Combine(home_dir, ".config"));
-			if (!FileSystem::DirectoryExists(config_dir.c_str()))
-				FileSystem::CreateDirectoryPath(config_dir.c_str(), false);
-
-			DataRoot = Path::Combine(config_dir, "PCSX2");
-		}
+		DataRoot = Path::Combine(xdg_config_home, "ArchyPie/configs/pcsx2");
 	}
 #elif defined(__APPLE__)
 	static constexpr char MAC_DATA_DIR[] = "Library/Application Support/PCSX2";
