--- a/src/filesys.cpp	2022-08-11 00:43:29.135494000 +0100
+++ b/src/filesys.cpp	2022-08-11 01:26:18.471701798 +0100
@@ -284,7 +284,7 @@
 		{
 			I_Error("Please set your HOME environment variable.\n");
 		}
-		configDir.Format("%s/.config/" GAME_DIR, home);
+		configDir.Format("%s/.config/archypie/ports/" GAME_DIR, home);
 	}
 	else
 		configDir.Format("%s/" GAME_DIR, xdg_config);
@@ -308,7 +308,7 @@
 		{
 			I_Error("Please set your HOME environment variable.\n");
 		}
-		documentsDir.Format("%s/.local/share/" GAME_DIR, home);
+		documentsDir.Format("%s/.config/archypie/ports/" GAME_DIR, home);
 	}
 	else
 		documentsDir.Format("%s/" GAME_DIR, xdg_data);
