--- a/atari800/src/cfg.c	2024-07-14 15:36:59.598786936 +0100
+++ b/atari800/src/cfg.c	2024-07-22 00:49:39.715129549 +0100
@@ -71,11 +71,11 @@
 /* If another default path config path is defined use it
    otherwise use the default one */
 #ifndef DEFAULT_CFG_NAME
-#define DEFAULT_CFG_NAME ".atari800.cfg"
+#define DEFAULT_CFG_NAME "/ArchyPie/configs/lr-atari800/lr-atari800.cfg"
 #endif
 
 #ifndef SYSTEM_WIDE_CFG_FILE
-#define SYSTEM_WIDE_CFG_FILE "/etc/atari800.cfg"
+#define SYSTEM_WIDE_CFG_FILE "/opt/archypie/configs/lr-atari800/lr-atari800.cfg"
 #endif
 
 #if defined (__PS3__) || defined(__PSL1GHT__)
