diff -u a/neo/sys/linux/main.cpp b/neo/sys/linux/main.cpp
--- a/neo/sys/linux/main.cpp	2022-06-13 02:36:23.000000000 +0100
+++ b/neo/sys/linux/main.cpp	2022-08-05 11:26:30.365502583 +0100
@@ -91,7 +91,7 @@
 	if (s)
 		D3_snprintfC99(save_path, sizeof(save_path), "%s/dhewm3", s);
 	else
-		D3_snprintfC99(save_path, sizeof(save_path), "%s/.local/share/dhewm3", getenv("HOME"));
+		D3_snprintfC99(save_path, sizeof(save_path), "%s/.config/archypie/ports/dhewm3", getenv("HOME"));
 }
 
 const char* Posix_GetExePath()
@@ -234,7 +234,7 @@
 		if (s)
 			idStr::snPrintf(buf, sizeof(buf), "%s/dhewm3", s);
 		else
-			idStr::snPrintf(buf, sizeof(buf), "%s/.config/dhewm3", getenv("HOME"));
+			idStr::snPrintf(buf, sizeof(buf), "%s/.config/archypie/ports/dhewm3", getenv("HOME"));
 
 		path = buf;
 		return true;
