--- a/BasiliskII/src/Unix/configure.ac 2021-01-23 22:21:57.989814552 -0500
+++ b/BasiliskII/src/Unix/configure.ac 2021-01-23 22:33:55.409812229 -0500
@@ -485,10 +485,6 @@
   AM_PATH_GTK(1.2.0, [
     GUI_CFLAGS="$GTK_CFLAGS"
     GUI_LIBS="$GTK_LIBS"
-    dnl somehow, <gnome-i18n.h> would redefine gettext() to nothing if
-    dnl ENABLE_NLS is not set, thusly conflicting with C++ <string> which
-    dnl includes <libintl.h>
-    AM_GNU_GETTEXT([external])
     B2_PATH_GNOMEUI([
       AC_DEFINE(HAVE_GNOMEUI, 1, [Define if libgnomeui is available.])
       GUI_CFLAGS="$GUI_CFLAGS $GNOMEUI_CFLAGS"
--- a/BasiliskII/src/Unix/sshpty.c	2021-08-17 19:33:32.036242000 +0100
+++ b/BasiliskII/src/Unix/sshpty.c	2021-08-17 19:39:40.697074962 +0100
@@ -77,7 +77,6 @@
 #define mysignal signal
 #include <signal.h>
 
-/* Pty allocated with _getpty gets broken if we do I_PUSH:es to it. */
 #if defined(HAVE__GETPTY) || defined(HAVE_OPENPTY)
 #undef HAVE_DEV_PTMX
 #endif
@@ -187,13 +186,7 @@
 	 * Push the appropriate streams modules, as described in Solaris pts(7).
 	 * HP-UX pts(7) doesn't have ttcompat module.
 	 */
-	if (ioctl(*ttyfd, I_PUSH, "ptem") < 0)
-		error("ioctl I_PUSH ptem: %.100s", strerror(errno));
-	if (ioctl(*ttyfd, I_PUSH, "ldterm") < 0)
-		error("ioctl I_PUSH ldterm: %.100s", strerror(errno));
 #ifndef __hpux
-	if (ioctl(*ttyfd, I_PUSH, "ttcompat") < 0)
-		error("ioctl I_PUSH ttcompat: %.100s", strerror(errno));
 #endif
 #endif
 	return 1;
