diff -ru a/Source/Core/Common/CommonPaths.h b/Source/Core/Common/CommonPaths.h
--- a/Source/Core/Common/CommonPaths.h	2023-12-18 01:22:30.599885841 +0000
+++ b/Source/Core/Common/CommonPaths.h	2023-12-18 02:13:59.908447031 +0000
@@ -19,7 +19,7 @@
 #elif defined(ANDROID)
 #define NORMAL_USER_DIR "/sdcard/dolphin-emu"
 #else
-#define NORMAL_USER_DIR "dolphin-emu"
+#define NORMAL_USER_DIR "dolphin"
 #endif
 #else  // ifndef STEAM
 #ifdef _WIN32
diff -ru a/Source/Core/UICommon/UICommon.cpp b/Source/Core/UICommon/UICommon.cpp
--- a/Source/Core/UICommon/UICommon.cpp	2023-12-18 01:22:30.745888976 +0000
+++ b/Source/Core/UICommon/UICommon.cpp	2023-12-18 23:54:58.852258251 +0000
@@ -448,32 +448,7 @@
 #else
     else
     {
-      user_path = home_path + "." NORMAL_USER_DIR DIR_SEP;
-
-      if (!File::Exists(user_path))
-      {
-        const char* data_home = getenv("XDG_DATA_HOME");
-        std::string data_path =
-            std::string(data_home && data_home[0] == '/' ? data_home :
-                                                           (home_path + ".local" DIR_SEP "share")) +
-            DIR_SEP NORMAL_USER_DIR DIR_SEP;
-
-        const char* config_home = getenv("XDG_CONFIG_HOME");
-        std::string config_path =
-            std::string(config_home && config_home[0] == '/' ? config_home :
-                                                               (home_path + ".config")) +
-            DIR_SEP NORMAL_USER_DIR DIR_SEP;
-
-        const char* cache_home = getenv("XDG_CACHE_HOME");
-        std::string cache_path =
-            std::string(cache_home && cache_home[0] == '/' ? cache_home : (home_path + ".cache")) +
-            DIR_SEP NORMAL_USER_DIR DIR_SEP;
-
-        File::SetUserPath(D_USER_IDX, data_path);
-        File::SetUserPath(D_CONFIG_IDX, config_path);
-        File::SetUserPath(D_CACHE_IDX, cache_path);
-        return;
-      }
+      user_path = home_path + "ArchyPie" DIR_SEP "configs" DIR_SEP NORMAL_USER_DIR DIR_SEP;
     }
 #endif
   }
