diff -ru a/third_party/physfs/src/physfs_platform_unix.c b/third_party/physfs/src/physfs_platform_unix.c
--- a/third_party/physfs/src/physfs_platform_unix.c	2022-12-27 14:41:18.047891454 +0000
+++ b/third_party/physfs/src/physfs_platform_unix.c	2022-12-27 14:42:36.553555413 +0000
@@ -351,13 +351,13 @@
         /* You end up with "$HOME/.local/share/Game Name 2" */
         envr = __PHYSFS_getUserDir();
         BAIL_IF_ERRPASS(!envr, NULL);  /* oh well. */
-        append = ".local/share/";
+        append = "ArchyPie/configs/vvvvvv/";
     } /* if */

-    len = strlen(envr) + strlen(append) + strlen(app) + 2;
+    len = strlen(envr) + strlen(append) + 1;
     retval = (char *) allocator.Malloc(len);
     BAIL_IF(!retval, PHYSFS_ERR_OUT_OF_MEMORY, NULL);
-    snprintf(retval, len, "%s%s%s/", envr, append, app);
+    snprintf(retval, len, "%s%s/", envr, append);
     return retval;
 } /* __PHYSFS_platformCalcPrefDir */

