--- a/menu/menu_driver.c	2022-05-07 10:30:51.308137000 +0100
+++ b/menu/menu_driver.c	2022-05-07 18:48:33.062688954 +0100
@@ -7961,7 +7961,7 @@
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         // menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
