diff -ru a/Src/OSD/Unix/FileSystemPath.cpp b/Src/OSD/Unix/FileSystemPath.cpp
--- a/Src/OSD/Unix/FileSystemPath.cpp	2024-06-21 17:06:11.789387869 +0100
+++ b/Src/OSD/Unix/FileSystemPath.cpp	2024-06-21 18:35:56.375276966 +0100
@@ -79,80 +79,17 @@
         }
 
         // Get user's HOME directory
-        if (pwd)
-        {
-            homePath = pwd->pw_dir;
-        }
-        else
-        {
-            homePath = getenv("HOME");
-        }
+        homePath = getenv("HOME");
 
-        // If Config path exists in current directory or the user doesn't have a HOME directory use current directory
-        if (FileSystemPath::PathExists("Config") || homePath.empty())
-        {
-            // Use current directory
-            if (pathType == Screenshots || pathType == Log)
-            {
-                finalPath = "";
-            }
-            else
-            {
-                // If directory doesn't exist, create it
-                if (!FileSystemPath::PathExists(strPathType))
-                {
-                    mkdir(strPathType.c_str(), 0775);
-                }
-                finalPath = strPathType;
-            }
-        }
-        // Check if $HOME/.supermodel exists
-        else if (FileSystemPath::PathExists(Util::Format() << homePath << "/.supermodel"))
+        if (FileSystemPath::PathExists(Util::Format() << homePath << "/ArchyPie/configs/supermodel-sinden"))
         {
-            // Use $HOME/.supermodel
-            finalPath = Util::Format() << homePath << "/.supermodel/" << strPathType;
+            finalPath = Util::Format() << homePath << "/ArchyPie/configs/supermodel-sinden/" << strPathType;
             // If directory doesn't exist, create it
             if (!FileSystemPath::PathExists(finalPath))
             {
                 mkdir(finalPath.c_str(), 0775);
             }
         }
-        // On Linux one may want to follow the XDG base directory specs (https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)
-        else
-        {
-            // Use $HOME/.config/supermodel or $HOME/.local/share/supermodel depending on the file type
-            if (pathType == Config)
-            {
-                finalPath = Util::Format() << homePath << "/.config/supermodel";
-                // If directory doesn't exist, create it
-                if (!FileSystemPath::PathExists(finalPath))
-                {
-                    mkdir(finalPath.c_str(), 0775);
-                }
-                // If directory doesn't exist, create it
-                finalPath = Util::Format() << homePath << "/.config/supermodel/Config";
-                if (!FileSystemPath::PathExists(finalPath))
-                {
-                    mkdir(finalPath.c_str(), 0775);
-                }
-            }
-            else
-            {
-                finalPath = Util::Format() << homePath << "/.local/share/supermodel";
-                // If directory doesn't exist, create it
-                if (!FileSystemPath::PathExists(finalPath))
-                {
-                    mkdir(finalPath.c_str(), 0775);
-                }
-                // If directory doesn't exist, create it
-                finalPath = Util::Format() << homePath << "/.local/share/supermodel/" << strPathType;
-                if (!FileSystemPath::PathExists(finalPath))
-                {
-                    mkdir(finalPath.c_str(), 0775);
-                }
-            }
-            
-        }
         
         if (finalPath != "")
         {
